src/App.tsx

```diff
@@ -1,39 +1,60 @@
 import React from 'react';
 import './App.scss';
+import { Clock } from './Clock';
+
+type State = {
+  hasClock: boolean,
+  clockName: string,
+};
 
 function getRandomName(): string {
   const value = Date.now().toString().slice(-4);
 
   return `Clock-${value}`;
 }
 
-export const App: React.FC = () => {
-  const today = new Date();
-  let clockName = 'Clock-0';
+export class App extends React.Component {
+  state: State = {
+    hasClock: true,
+    clockName: 'Clock-0',
+  };
+
+  componentDidMount(): void {
+    window.setInterval(() => {
+      this.setState({ clockName: getRandomName() });
+    }, 3300);
+
+    window.addEventListener('contextmenu', this.hideClock);
+    window.addEventListener('click', this.showClock);
+  }
 
-  // This code starts a timer
-  const timerId = window.setInterval(() => {
-    clockName = getRandomName();
-  }, 3300);
+  componentWillUnmount(): void {
+    window.removeEventListener('contextmenu', this.hideClock);
+    window.removeEventListener('click', this.showClock);
```

Comment:
Maybe it's a good idea not only to remove the event listeners but also clear interval here

Reply:
agree with [comment](https://github.com/mate-academy/react_clock/pull/1672/files#r1269991997)